tests/manage/mcg/test_bucket_policy.py

```diff
@@ -264,6 +265,16 @@ def test_object_actions(self, mcg_obj, bucket_factory):
         get_policy = get_bucket_policy(mcg_obj, obc_obj.bucket_name)
         logger.info(f"Got bucket policy: {get_policy['Policy']}")
 
+        # Put object using OBC account(as an Admin)
+        if float(config.ENV_DATA['ocs_version']) >= 4.6:
```

Comment:
I feel that this fix tailors the test to work with 4.6, rather than fixing the fundamental issues of it.

The current flow is flawed:
https://github.com/red-hat-storage/ocs-ci/pull/3018/files#diff-4b4e7d3ba98ac96bbc8edf7387c285d2R218
In this part, there's an attempt to get the object. The get should be denied. However, if it succeeds, the test continues executing just the same, which shouldn't be the case - the test should fail.

Same here:
https://github.com/red-hat-storage/ocs-ci/pull/3018/files#diff-4b4e7d3ba98ac96bbc8edf7387c285d2R237
The deletion should fail, but it succeeds, and it doesn't fail the test - even though it should.

True - reuploading the object before trying to `get` it solves the problem - but on the other hand, the test simply doesn't represent the true state of things in 4.6. I think it'll be better to rething the flow, or at least add verifications that actions that should fail, have failed successfully.

WDYT, @parikshithb?

Reply:
> I feel that this fix tailors the test to work with 4.6, rather than fixing the fundamental issues of it.
> 
> The current flow is flawed:
> https://github.com/red-hat-storage/ocs-ci/pull/3018/files#diff-4b4e7d3ba98ac96bbc8edf7387c285d2R218
> In this part, there's an attempt to get the object. The get should be denied. However, if it succeeds, the test continues executing just the same, which shouldn't be the case - the test should fail.
> 
> Same here:
> https://github.com/red-hat-storage/ocs-ci/pull/3018/files#diff-4b4e7d3ba98ac96bbc8edf7387c285d2R237
> The deletion should fail, but it succeeds, and it doesn't fail the test - even though it should.
> 

I am not sure I quite understand, both GET and DELETE(line 218 & 237) actions does fail right( i.e Access denied)? Both the actions are denied to obc.account principal according to the policy set. 

> True - reuploading the object before trying to `get` it solves the problem - but on the other hand, the test simply doesn't represent the true state of things in 4.6. I think it'll be better to rething the flow, or at least add verifications that actions that should fail, have failed successfully.
> 
The new code is just for noobaa s3 access principal (i.e user.email) where I am uploading an object from obc.account(who is the admin) for version > 4.6 and verifying the policy set to user.email. 

There might be better way than re uploading, so can you please explain more about the flow? 
> WDYT, @parikshithb?

Reply:
My bad, i missed that for obc.account, actions were not actually denied. 

What about this flow: 
1. We provide **user** principal as well along with **obc_obj.obc_account**.

        bucket_policy_generated = gen_bucket_policy(
            user_list=[obc_obj.obc_account, user.email],
            actions_list=['PutObject'],
            resources_list=[f'{obc_obj.bucket_name}/{"*"}']
        )

2. Then try-catch like this for both **get** and **delete** actions: 

        try:
            if float(config.ENV_DATA['ocs_version']) >= 4.6:
                s3_get_object(user, obc_obj.bucket_name, object_key)
            else:
                s3_get_object(obc_obj, obc_obj.bucket_name, object_key)
        except boto3exception.ClientError as e:
            logger.info(e.response)
            response = HttpResponseParser(e.response)
            if response.error['Code'] == 'AccessDenied':
                logger.info('Get Object action has been denied access')
            else:
                raise UnexpectedBehaviour(f"{e.response} received invalid error code {response.error['Code']}")

Reply:
@MeridianExplorer Changed the flow.